packages/Meshfree/test/tstMeshfreeComparison.cpp

```diff
@@ -0,0 +1,271 @@
+
+constexpr int DIM = 3;
+
+template <typename DeviceType>
+struct Helper
+{
+    static std::vector<std::array<double, DIM>>
+    makeGridPoints( std::array<int, DIM> const &n_points,
+                    std::array<double, DIM> const &lower_corner,
+                    std::array<double, DIM> const &upper_corner )
+    {
+        static_assert( DIM == 3, "Assume three dimensional geometry" );
+
+        std::vector<std::array<double, DIM>> grid_points;
+
+        for ( int i = 0; i < n_points[0]; ++i )
+        {
+            for ( int j = 0; j < n_points[1]; ++j )
+            {
+                for ( int k = 0; k < n_points[2]; ++k )
+                {
+                    std::array<double, DIM> point = {
+                        lower_corner[0] +
+                            ( upper_corner[0] - lower_corner[0] ) /
+                                ( n_points[0] + 1 ) * ( i + 1 ),
+                        lower_corner[1] +
+                            ( upper_corner[1] - lower_corner[1] ) /
+                                ( n_points[1] + 1 ) * ( j + 1 ),
+                        lower_corner[2] +
+                            ( upper_corner[2] - lower_corner[2] ) /
+                                ( n_points[2] + 1 ) * ( k + 1 )};
+                    grid_points.push_back( point );
+                }
+            }
+        }
+
+        return grid_points;
+    }
+
+    static Kokkos::View<DataTransferKit::Coordinate **, DeviceType>
+    makePoints( std::vector<std::array<double, DIM>> const &in )
+    {
+        int const n = in.size();
+        Kokkos::View<DataTransferKit::Coordinate **, DeviceType> out( "points",
+                                                                      n, DIM );
+        auto out_host = Kokkos::create_mirror_view( out );
+        for ( int i = 0; i < n; ++i )
+            for ( int j = 0; j < DIM; ++j )
+                out_host( i, j ) = in[i][j];
+        Kokkos::deep_copy( out, out_host );
+        return out;
+    }
+
+    static Kokkos::View<double *, DeviceType>
+    makeValues( std::vector<double> const &in )
+    {
+        int const n = in.size();
+        Kokkos::View<double *, DeviceType> out( "points", n );
+        auto out_host = Kokkos::create_mirror_view( out );
+        for ( int i = 0; i < n; ++i )
+            out_host( i ) = in[i];
+        Kokkos::deep_copy( out, out_host );
+        return out;
+    }
+};
+
+template <typename Operator>
+void testOperator( int source_points_per_dim, int target_points_per_dim )
+{
+    using DeviceType = typename Operator::device_type;
+
+    MPI_Comm comm = MPI_COMM_WORLD;
+    int comm_rank;
+    MPI_Comm_rank( comm, &comm_rank );
+
+    int comm_size;
+    MPI_Comm_size( comm, &comm_size );
+
+    int l = std::ceil( std::cbrt( comm_size ) );
+
+    double offset_x = ( comm_rank % l ) * 1. / l;
+    double offset_y = ( ( comm_rank / l ) % l ) * 1. / l;
+    double offset_z = ( ( comm_rank / ( l * l ) ) % l ) * 1. / l;
```

Comment:
I'm not sure what's going on here.

Reply:
That should be more or less the same code we use in the distributed driver in ArborX. For determining the local subboxes (and we scale the overall box to the unit box).